snakemake_wrapper_utils/java.py

```diff
@@ -1,21 +1,33 @@
+import sys
+
 def get_java_opts(snakemake):
     """Obtain java_opts from params, and handle resource definitions in resources."""
 
     java_opts = snakemake.params.get("java_opts", "")
     # Getting memory in megabytes, if java opts is not filled with -Xmx parameter
     # By doing so, backward compatibility is preserved
-    if "mem_mb" in snakemake.resources.keys() and not "-Xmx" in java_opts:
+    if "mem_mb" in snakemake.resources.keys():
+        if "-Xmx" in java_opts:
+            sys.exit("You have specified resources.mem_mb and provided `-Xmx` in params.java_opts. For Java memory specifications, please only use resources.mem_mb or resources.mem_gb in the future.")
```

Comment:
Only mem_mb should be recommended.

Reply:
But I really like `mem_gb` for many use cases (and @tdayris implemented `mem_gb` upon my suggestion for the Java tools wrappers). Once I start requesting certain amounts of memory, this usually implies that I am dealing with GB of RAM. And it reduces the burden of (mentally) dividing / multiplying by 1024 (JVM deals in Mebibytes and Gibibytes).

Reply:
I do see the point that within a workflow you should only use either `mem_mb` or `mem_gb` and only offering one is a way to force the users to one. But I'd argue that `mem_gb` might even be the better default with today's tools and machines.

Reply:
Indeed, it might be more convenient in many cases. But 1GB can also be far too much for certain tools. The thing is that this is passed on to cluster profiles and cloud executors, and all of those just understand mem_mb. However, we could modify Snakemake to always infer mem_mb from mem_gb. So, let's do the following. For now, just mention mem_mb. Once Snakemake is modified accordingly, also mention mem_gb.

Reply:
OK, deal! :D

Reply:
P.S.: It seems like I could merge even without an approving review. Maybe adjust the repo config?